### PR TITLE
Add appliedmanifestwork webhook

### PR DESCRIPTION
**Pull request by qiujian16**

### CODE REVIEW

File: pkg/webhook/appliedmanifestwork_webhook_test.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+				Operation: admissionv1beta1.Create,
+			},
+			hubHash:                 "test",
+			appliedManifestWorkName: "test-work-0",
+			expectedResponse: &admissionv1beta1.AdmissionResponse{
+				Allowed: true,
+			},
+		},
+		{
+			name: "validate creating AppliedManaifestWork with invalid name",
```
Comment (pmorie) [inline pkg/webhook/appliedmanifestwork_webhook_test.go:57]: typy: Manaif

File: pkg/webhook/appliedmanifestwork_webhook_test.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+				},
+			},
+			hubHash:                 "test",
+			appliedManifestWorkName: "test-work-0",
+			expectedResponse: &admissionv1beta1.AdmissionResponse{
+				Allowed: true,
+			},
+		},
+		{
+			name: "validate creating AppliedManaifestWork",
```
Comment (pmorie) [inline pkg/webhook/appliedmanifestwork_webhook_test.go:45]: typo

File: pkg/webhook/appliedmanifestwork_webhook_test.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+			hubHash:                 "test",
+			appliedManifestWorkName: "test-work-1",
+			expectedResponse: &admissionv1beta1.AdmissionResponse{
+				Allowed: false,
+				Result: &metav1.Status{
+					Status: metav1.StatusFailure, Code: http.StatusBadRequest, Reason: metav1.StatusReasonBadRequest,
+					Message: "invalid appliedmanifestwork name",
+				},
+			},
+		},
```
Comment (pmorie) [inline pkg/webhook/appliedmanifestwork_webhook_test.go:87]: would you add a test case for invalid payload?

**Review by pmorie — COMMENTED** [review]
Looking really good, just a couple notes

File: deploy/webhook/appliedmanifestwork_webhook.yaml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  - operations:
+    - CREATE
+    - UPDATE
+    apiGroups:
+    - work.open-cluster-management.io
+    apiVersions:
+    - "*"
+    resources:
+    - appliedmanifestworks
+  admissionReviewVersions: ["v1beta1"]
```
Comment (RainbowMango) [inline deploy/webhook/appliedmanifestwork_webhook.yaml:24]: Why not use `v1` for `admissionReviewVersions`?

Comment (qiujian16) [reply deploy/webhook/appliedmanifestwork_webhook.yaml:24]: yes, i think we could use v1 :) will update that

**Pull request closed without merging**